driver-core/src/main/com/mongodb/client/model/expressions/BooleanExpression.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.model.expressions;
+
+/**
+ * Expresses a boolean value.
+ */
+public interface BooleanExpression extends Expression {
+
+    /**
+     * Returns logical true if this expression evaluates to logical false.
```

Comment:
I think the Javadoc should be written in terms of expressions rather than values.  So in this case something like:

`Returns an expression representing the logical negation of this {@code BooleanExpression}.`

Reply:
This is turning out a bit more difficult to formulate at this stage than I thought it would. I'd like to defer docs until a later PR.

Reply:
Acknowledged, but please open a JAVA ticket for it.

Reply:
https://jira.mongodb.org/browse/JAVA-4799